### PR TITLE
SUIT: fail fetch if the image size doesn't match expected

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This adds a few more checks to the fetch code to ensure that the fetched binary matched the size given in the manifest.

### Testing procedure

Verify the workflow with `examples/suit_update`

### Issues/PRs references

None

### CODE REVIEW

File: sys/suit/handlers_command_seq.c [hunk 1]
```diff
@@ -375,7 +375,9 @@ static int _dtv_verify_image_match(suit_manifest_t *manifest, int key,
     if (res != 0) {
         return SUIT_ERR_COND;
     }
-    riotboot_flashwrite_finish(manifest->writer);
+    if (!IS_ACTIVE(MODULE_SUIT_TRANSPORT_MOCK)) {
+        riotboot_flashwrite_finish(manifest->writer);
+    }
```
Comment (fjmolinas) [inline sys/suit/handlers_command_seq.c]: Maybe a comment would be as useful as the commit message here? (your call)

**Comment by fjmolinas** [comment]
Tested all OK

```
suit_coap: started.
Starting the shell
> ifconfig
ifconfig
Iface  4  HWaddr: 22:62:23:22:5B:67
          L2-PDU:1500  MTU:1500  HL:64  RTR
          Source address length: 6
          Link type: wired
          inet6 addr: fe80::2062:23ff:fe22:5b67  scope: link  VAL
pinging node...
PING fe80::2062:23ff:fe22:5b67%riot0(fe80::2062:23ff:fe22:5b67%riot0) 56 data bytes

--- fe80::2062:23ff:fe22:5b67%riot0 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 43.923/43.923/43.923/0.000 ms
pinging node succeeded.
TEST PASSED
```

Squash right away if you address my comment @bergzand

File: sys/suit/handlers_command_seq.c (hunk 1 above)
Comment (bergzand) [reply sys/suit/handlers_command_seq.c]: Squashed in a comment!

**Comment by bergzand** [comment]
@fjmolinas All green here! :)

**Review by fjmolinas — APPROVED** [review]
ACK and go!

**Comment by bergzand** [comment]
Thanks!

**Pull request merged**